src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -399,7 +405,13 @@ public override EnvDTE.FileCodeModel GetFileCodeModel(DocumentId documentId)
                 this.EnsureEditableDocuments(changedDocs);
             }
 
-            return base.TryApplyChanges(newSolution, progressTracker);
+            if (base.TryApplyChanges(newSolution, progressTracker))
+            {
+                _refactorNotifyServices.TryNotifyChangesSynchronously(this, newSolution, currentSolution, _foregroundObject.ThreadingContext);
+                return true;
+            }
```

Comment:
this feels very expensive.  TryApplyChanges is a very significant codepath.  i'm not sure how i feel doing a large piece of work like this in here.  Can this not be done by the feature instead?